### PR TITLE
chore: only tag/push latest Docker for latest semantic tag

### PR DESCRIPTION
**Pull request by markphelps**
This _should_ fix the issue where creating a backport release was incorrectly also tagging the image as `latest` and pushing to our Docker registries

This removes the tagging of latest from `goreleaser` workflow and instead tags and pushes the image only if it is actually the most recent semantic release.

I tested this locally by writing a little script:

```bash
# test.sh

if [[ $(git tag -l --sort=-version:refname | head -n 1) == $TAG ]]; then
    echo "MATCH"
else
    echo "NO MATCH"
fi
```

```console
$ ./test.sh
NO MATCH

$ TAG=v1.15.1 ./test.sh
NO MATCH

$ TAG=v1.15.2 ./test.sh
MATCH
```

### CODE REVIEW

File: .goreleaser.yml [hunk 1]
```diff
@@ -83,26 +85,16 @@ dockers:
...
 docker_manifests:
-  - name_template: "{{ if .IsNightly }}flipt/flipt:nightly{{ else }}flipt/flipt:latest{{ end }}"
-    image_templates:
-      - "{{ if .IsNightly }}flipt/flipt:{{ incpatch .Version }}-nightly-amd64{{ else }}flipt/flipt:{{ .Tag }}-amd64{{ end }}"
-      - "{{ if .IsNightly }}flipt/flipt:{{ incpatch .Version }}-nightly-arm64{{ else }}flipt/flipt:{{ .Tag }}-arm64{{ end }}"
   - name_template: "{{ if .IsNightly }}flipt/flipt:{{ incpatch .Version }}-nightly{{ else }}flipt/flipt:{{ .Tag }}{{ end }}"
     image_templates:
       - "{{ if .IsNightly }}flipt/flipt:{{ incpatch .Version }}-nightly-amd64{{ else }}flipt/flipt:{{ .Tag }}-amd64{{ end }}"
       - "{{ if .IsNightly }}flipt/flipt:{{ incpatch .Version }}-nightly-arm64{{ else }}flipt/flipt:{{ .Tag }}-arm64{{ end }}"
-  - name_template: "{{ if .IsNightly }}markphelps/flipt:nightly{{ else }}markphelps/flipt:latest{{ end }}" # TODO: deprecate
```
Comment (markphelps) [inline .goreleaser.yml:94]: gah, this does break the `nightly` tag tho..

Comment (markphelps) [reply .goreleaser.yml:94]: I think I fixed it by breaking out nightly into its own `.goreleaser.nightly.yml` and using the [includes](https://goreleaser.com/customization/includes/?h=include) functionality of goreleaser

**Comment by GeorgeMac** [comment]
I've been seeing this block on cockroach a lot lately. This is an attempt to see if waiting for SQL helps: https://github.com/flipt-io/flipt/pull/1183

**Review by GeorgeMac — APPROVED** [review]
Nice.

**Pull request merged**